src/rpc/rpcmapper.rs

```diff
@@ -75,26 +67,35 @@ impl RpcMapper {
     ///
     /// Returns a `Future` that resolves to a `Result` either containing the expected return type of the RPC method wrapped,
     /// or an [`RpcMapperError`].
-    pub fn map_response<T: prost::Message + Default>(
-        response_future: Pin<Box<dyn Future<Output = RpcClientResult>>>,
-    ) -> Pin<Box<dyn Future<Output = Result<T, RpcMapperError>>>> {
-        Box::pin(async move {
-            match response_future.await {
-                Ok(payload) => {
-                    // we got some useable response payload
-                    match payload.to_any() {
-                        // we were able to extract an Any from the response_future
-                        Ok(any) => match T::decode(any.value.as_slice()) {
-                            Ok(result) => Ok(result), // expected response type could be decoded
-                            Err(error) => Err(RpcMapperError::InvalidPayload(error.to_string())), // ... or not
-                        },
-                        // ... or not
-                        Err(error) => Err(RpcMapperError::ProtobufError(error.to_string())),
+    pub async fn map_response<T, F>(response_future: F) -> Result<T, RpcMapperError>
+    where
+        T: prost::Message + Default,
+        F: Future<Output = RpcClientResult>,
+    {
+        match response_future.await {
+            Ok(payload) => {
+                // we got some useable response payload
+                let any: Any = payload.into();
+                if any != Any::default() {
+                    match T::decode(any.value.as_slice()) {
```

Comment:
This match could be replaced with a [map_err](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err).

Reply:
done